core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala

```diff
@@ -116,6 +108,37 @@ import com.datastax.driver.core.utils.UUIDs
   }
 
   private case object QueryPoll
+
+  private case class StageState(
+    state: QueryState,
+    // updated each time a new row is read, may skip delayed events with lower offsets
+    // these gaps will be detected for the same pid, for cross pid delayed events it
+    // relies on the eventual consistency delay
+    fromOffset: UUID,
+    // upper limit for next. Always kept at least the eventual consistency in the past
+    toOffset: UUID,
+    // for each persistence id what is the latest tag pid sequenceNr number and offset
+    tagPidSequenceNrs: Map[PersistenceId, (TagPidSequenceNr, UUID)],
+    missingLookup:     Option[LookingForMissing],
+    bucketSize:        BucketSize) {
+
+    val isLookingForMissing: Boolean = missingLookup.isDefined
+
+    lazy val currentTimeBucket: TimeBucket = TimeBucket(fromOffset, bucketSize)
```

Comment:
Is it really a gain to make it lazy (with the synchronization cost) vs just making it a `val` or `def`? Looks like it is touched in a bunch of places anyway?